corehq/apps/app_manager/suite_xml/sections/entries.py

```diff
@@ -545,6 +550,18 @@ def add_remote_query_datums(self, datums):
                 result.append(datum)
         return result
 
+    def rename_datum_nodeset_for_query(self, datum, query_datum):
+        """Rename the instance in the case datum to match the instance used by the query datum
+        """
+        instance_name = query_datum.datum.storage_instance
+        if instance_name == "results":
+            return datum
+        instance_pattern = r"""instance\(['"]results:(.*?)['"]\)"""
+        new_instance = f"instance('{instance_name}')"
+        renamed = re.sub(instance_pattern, new_instance, datum.datum.nodeset)
```

Comment:
know where these `datum`s are coming from?  Is it possible to instead have them use the correct instance name up front rather than modify that later?  Similarly, is there any risk associated with mutating the datums?  If they're accessed elsewhere, the instance name used will depend on whether that's called before or after this.

Reply:
the `datum` are created via [get_case_datums_basic_module](https://github.com/dimagi/commcare-hq/blob/d35070e90dbcad8d231be216e0d913e9279b3e84/corehq/apps/app_manager/suite_xml/sections/entries.py#L518) which generates/returns all the datum within the <session> block (for that module/form) During the datum creation, it seems we do know whether the module is using inline_case_search so I was able to name it upfront [d35070e](https://github.com/dimagi/commcare-hq/pull/33635/commits/d35070e90dbcad8d231be216e0d913e9279b3e84)